bootstrap/config/overlays/ksonnet/kfctl_default.yaml

```diff
@@ -4,10 +4,37 @@ metadata:
   name: config
 spec:
   componentParams:
+    argo:
```

Comment:
we'll no longer use ksonnet in about a week. We're waiting on a few blocking PRs. 
injectIstio will also be replaced by an istio overlay and the istio resources under dependencies/istio will be deployed from manifests. So most of the changes in here will be reverted. 

If these changes stabilize main then it's a good reason to merge them, even though most of the changes won't be used after PRs 3045, 3375, and manifests 102. Let me know if that's the goal with this PR.

Reply:
@kkasravi to my understanding, `kfctl` can support multiple package managers.
Since that seems to be the case, isn't it ok if we stabilize `ksonnet` and just leave it as is?

Reply:
@yanniszark i think this would be difficult, having seen what happens (or doesnt' happen) between kustomize and ksonnet thus far. It wouldn't really be 'deprecated' since there would be differences in what kustomize installs and what ksonnet installs. My feeling is just getting the kustomize testing to parity with ksonnet and standardizing best practices within manifests is a higher priority than making sure ksonnet is working correctly, IMO it ends up diverting efforts towards switching to kustomize.

Reply:
@kkasravi Hey Kam, I understand your concern about getting everything working with kustomize, and I agree. We would have made the changes directly on the kustomize package manager, if we had a solid place to start from.

But given that there is currently no support for Istio on-prem, and it is a stated goal for 0.6 that we want to support Istio uniformly and not have it as a GCP-only feature, we need to have a way to iterate now, to make it for 0.6. We need to have a solid way to use the kfctl tool not only on GCP but also on-prem, and I understand kfctl + kustomize are not yet stable in this area.

Can we get this PR merged, so we have solid ground to work on now, during the 0.6 cycle? Even it means using kfctl + ksonnet. This will allow us to move in parallel, while kfctl + kustomize stabilizes. We are happy to contribute fixes and ensure everything works with kustomize as well, but I understand the current evel of kfctl + kustomize support does not allow us to do so.

Reply:
thanks @vkoukis i believe this may address istio on-prem, but may also break when '--platform gcp'.
I've added comments in specific areas.

Reply:
@vkoukis Can you clarify what you mean by 

>  Istio uniformly and not have it as a GCP-only feature, 

As I mentioned in slack after thinking about it following the community meeting I do think it might make sense for ISTIO installation to be tied to the platform and the responsibility of whoever owns the corresponding platform.

ISTIO is heavily tied to the platform; IMO.

The only other support needed for ISTIO should be the definition of virtual services and those are all platform agnostic.